### ISSUE TITLE
[Accessibility]: Need better visual feedback when toggling smart search

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43128]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

Note: smart search must be enabled by adding the setting `defaultPatternType: lucky`

When toggling Smart Search, there should be a delay after clicking on the radio button or using arrow keys so that the popup doesn't close immediately. Right now, especially when using the keyboard, the process is very confusing and it looks like nothing happened unless you notice that the (comparatively small) smart search icon changed color.the 

### Expected behavior

There should be a small delay between when changing the smart search radio button and the dialog closing

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard
